### PR TITLE
fix(LT-3991): moved in audit abstractions

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Snow.Audit/Abstractions/IAuditModel.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        public DateTime Timestamp { get; set; }
+
+        public string CorrelationId { get; set; }
+
+        public string UserName { get; set; }
+
+        public AuditEventType Type { get; set; }
+
+        public T DataType { get; set; }
```
Comment (gponomarev-lykke) [inline src/Lykke.Snow.Audit/Abstractions/IAuditModel.cs:21]: I think Object or Value would be a better name
As I understand, this property represents a real object, not its type or class
If it's really the type of the object, perhaps a constraint could be added to T?

File: src/Lykke.Snow.Audit/Abstractions/IAuditableObject.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{
+    /// <summary>
+    /// Interface to be implemented by the object that is auditable
+    /// </summary>
+    /// <typeparam name="T">The type of the audit data</typeparam>
+    public interface IAuditableObject<out T>
+    {
+        T GetAuditDataType();
+
+        string GetAuditReference();
```
Comment (gponomarev-lykke) [inline src/Lykke.Snow.Audit/Abstractions/IAuditableObject.cs:11]: Please add some xml comments on how to implement methods of this interface, e.g. it's not clear from the code what AuditReference is meant to be.

File: src/Lykke.Snow.Audit/Abstractions/IAuditModel.cs (hunk 1 above)
Comment (tarurar) [reply src/Lykke.Snow.Audit/Abstractions/IAuditModel.cs:21]: It is an enum which is relevant to the domain service only and designates the type of the auditable object. For example: https://bitbucket.org/lykke-snow/lykke.snow.mdm/src/master/src/Lykke.Snow.Mdm.Core/Models/Enums/AuditDataType.cs

File: src/Lykke.Snow.Audit/Abstractions/IAuditableObject.cs (hunk 2 above)
Comment (tarurar) [reply src/Lykke.Snow.Audit/Abstractions/IAuditableObject.cs:19]: updated

**Review by lykke-vashetsin — APPROVED** [review]

**Pull request merged**